SpokeStack/SpokeStack/WakeWordSpeechRecognizer.swift

```diff
@@ -0,0 +1,660 @@
+
+        /// Decode the FFT outputs into the filter model's input
+        /// Compute the nagitude (abs) of each complex stft component
+        /// The first and last stft components contain only real parts
+        /// and are stored in the first of the first two positions of the stft
+        /// output. The remaining components contact real / imaginary parts
+        
+        
+        let frameCount: Int = (self.fftFrame.count / 2)
+        var components: Array<Float> = []
+        components.reserveCapacity(frameCount)
+        
+        /// Populate the components
+    
+        let firstComponent: Float = self.fftFrame.first!
+        components.append(firstComponent)
+    
+        var i: Int = 1
+        repeat {
+            
+            let re: Float = self.fftFrame[i * 2 + 0]
+            let im: Float = self.fftFrame[i * 2 + 1]
+            let ab: Float = sqrt(re * re + im * im)
+            
+            components.append(ab)
+            
+            i += 1
+            
+        } while i < frameCount
+    
+        let lastComponent: Float = self.fftFrame[1]
+        components.append(lastComponent)
+
+        /// Run the predictions
+    
+        guard let multiArray = try? MLMultiArray(shape: [257,1,1], dataType: .float32) else {
+                
+                fatalError("Unexpected runtime error. MLMultiArray")
+        }
+
+        for (index, value) in components.enumerated() {
+            multiArray[index] = NSNumber(value: value)
+        }
+    
+        do {
+        
+            let input: WakeWordFilterInput = WakeWordFilterInput(linspec_inputs__0: multiArray)
+            let predictions: WakeWordFilterOutput = try self.wwfilter.prediction(input: input)
+            
+            /// Copy the current mel frame into the mel window
+            
+            self.frameWindow.rewind().seek(self.melWidth)
+            
+            for i in 0..<ModelConstants.numOfMelOutputs {
+                
+                let result = String(describing: predictions.melspec_outputs__0[i])
+                try? self.frameWindow.write(predictions.melspec_outputs__0[i].floatValue)
+                
+//                print("what is my result from filter \(result)")
+            }
+            
+            /// Detect
+            
+            self.detect()
+            
+        } catch let modelFilterError {
+            
+            fatalError("modelFilterError is thrown \(modelFilterError)")
+        }
+    }
+    
+    private func detect() -> Void {
+        
+        print("detect is being fired off")
+
+        /// Transfer the mel filterbank window to the detector model's inputs
+        
+        self.frameWindow.rewind()
+        
+        guard let multiArray = try? MLMultiArray(shape: [1,40,40], dataType: .float32) else {
+            
+                fatalError("Unexpected runtime error. MLMultiArray")
+        }
+        
+        for index in 0..<ModelConstants.numOfMelOutputs {
```

Comment:
This loop is not copying the values from the frameWindow into the detector inputs, and you are only filling 40 values in the input array. I suspect this is why you are getting NaN values in your detector output, since the remainder of the array is not being initialized. This array should be filled with the contents of the frame window that was populated by the `filter` method. See https://github.com/pylon/spokestack-android/blob/master/src/main/java/com/pylon/spokestack/wakeword/WakewordTrigger.java#L461 for the correct implementation in the android repo, lines 463-466.